### PR TITLE
Group tests in a dedicated testing package

### LINKED ISSUE #620: Group tests in a dedicated testing package
# Problem

Our tests are in the root package now (`e2e`, `testing`).
I'd be nice to move them to dedicated test package.

# Implementation

The package name should be `t`, to make it stand out and short.
`e2e` should be moved to `t/e2e`.
`testing` should be renamed to `t/utils`

# Acceptance Criteria

Tests still run and pass for:
* `make test-unit`
* `make test-e2e`

### PR DESCRIPTION
**Pull request by sekl**
Moved `e2e` and `testing` directories to a new package `t`.

All I did was moving the files and updating imports and docs where necessary. Please let me know if this is as intended or if I missed anything. 

Closes #620

### CODE REVIEW

File: t/utils/utils.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package testing
+package t
```
Comment (mandrigin) [inline t/utils/utils.go]: that should be `package utils`, right?

File: t/e2e/whisper/whisper_jail_test.go [hunk 2]
```diff
@@ -6,10 +6,10 @@ import (
 
 	"github.com/ethereum/go-ethereum/crypto"
 	whisper "github.com/ethereum/go-ethereum/whisper/whisperv5"
-	"github.com/status-im/status-go/e2e"
 	"github.com/status-im/status-go/geth/common"
 	"github.com/status-im/status-go/static"
-	. "github.com/status-im/status-go/testing"
+	e2e "github.com/status-im/status-go/t/e2e"
```
Comment (mandrigin) [inline t/e2e/whisper/whisper_jail_test.go:11]: wouldn't it be exported as `e2e` anyway?

File: t/e2e/whisper/whisper_mailbox_test.go [hunk 3]
```diff
@@ -12,10 +13,10 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/p2p/discover"
 	"github.com/ethereum/go-ethereum/whisper/whisperv5"
-	"github.com/status-im/status-go/e2e"
 	"github.com/status-im/status-go/geth/api"
 	"github.com/status-im/status-go/geth/rpc"
-	. "github.com/status-im/status-go/testing"
+	e2e "github.com/status-im/status-go/t/e2e"
```
Comment (mandrigin) [inline t/e2e/whisper/whisper_mailbox_test.go:18]: wouldn't it be exported as `e2e` anyway?

File: t/e2e/whisper/mailservice_test.go [hunk 4]
```diff
@@ -6,9 +6,9 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/status-im/status-go/e2e"
 	"github.com/status-im/status-go/geth/node"
 	"github.com/status-im/status-go/geth/params"
+	e2e "github.com/status-im/status-go/t/e2e"
```
Comment (mandrigin) [inline t/e2e/whisper/mailservice_test.go:11]: wouldn't it be exported as `e2e` anyway?

**Review by mandrigin — CHANGES_REQUESTED** [review]
Thanks for your first PR!
Yes, the idea was to gather the test files together in a separate package.

There are a few questions to this PR.

File: t/utils/utils.go (hunk 1 above)
Comment (sekl) [reply t/utils/utils.go]: Yes, I've changed it, thanks!

File: t/e2e/whisper/mailservice_test.go (hunk 4 above)
Comment (sekl) [reply t/e2e/whisper/mailservice_test.go:11]: I thought so, but goimports keeps saving it like this and simple changing it and ignore goimports results in a failed test. Does it mean something else is wrong about the package structure, or can it stay like this?

Comment (mandrigin) [reply t/e2e/whisper/mailservice_test.go:11]: hmm, I don't see anything wrong with the package structure there...
ok, we can keep it as is then

Comment (sekl) [reply t/e2e/whisper/mailservice_test.go:11]: Great, let me know if I should change anything else then. :)

**Review by mandrigin — APPROVED** [review]

File: t/e2e/whisper/whisper_mailbox_test.go [hunk 5]
```diff
@@ -305,7 +306,7 @@ func (d *groupChatParams) Encode() (string, error) {
 
 //Start status node
 func (s *WhisperMailboxSuite) startBackend(name string) (*api.StatusBackend, func()) {
-	datadir := "../../.ethereumtest/mailbox/" + name
+	datadir := filepath.Join(RootDir, ".ethereumtest/mailbox/") + name
```
Comment (adambabik) [inline t/e2e/whisper/whisper_mailbox_test.go]: `filepath.Join(RootDir, ".ethereumtest/mailbox", name)` should work the same.

File: t/e2e/whisper/whisper_mailbox_test.go [hunk 6]
```diff
@@ -332,15 +333,15 @@ func (s *WhisperMailboxSuite) startMailboxBackend() (*api.StatusBackend, func())
...
 	mailboxConfig.WhisperConfig.Enabled = true
-	mailboxConfig.KeyStoreDir = "../../.ethereumtest/mailbox/mailserver"
+	mailboxConfig.KeyStoreDir = datadir
 	mailboxConfig.WhisperConfig.EnableMailServer = true
-	mailboxConfig.WhisperConfig.IdentityFile = "../../static/keys/wnodekey"
-	mailboxConfig.WhisperConfig.PasswordFile = "../../static/keys/wnodepassword"
-	mailboxConfig.WhisperConfig.DataDir = "../../.ethereumtest/mailbox/mailserver/data"
+	mailboxConfig.WhisperConfig.IdentityFile = filepath.Join(RootDir, "/static/keys/wnodekey")
+	mailboxConfig.WhisperConfig.PasswordFile = filepath.Join(RootDir, "/static/keys/wnodepassword")
+	mailboxConfig.WhisperConfig.DataDir = datadir + "/data"
```
Comment (adambabik) [inline t/e2e/whisper/whisper_mailbox_test.go]: `filepath.Join(datadir, "data")`?

**Review by adambabik — APPROVED** [review]

File: t/e2e/whisper/whisper_mailbox_test.go (hunk 6 above)
Comment (mandrigin) [reply t/e2e/whisper/whisper_mailbox_test.go]: yeah, let's use `Join` everywhere

File: t/e2e/whisper/whisper_mailbox_test.go (hunk 5 above)
Comment (mandrigin) [reply t/e2e/whisper/whisper_mailbox_test.go]: +1

**Review by mandrigin — CHANGES_REQUESTED** [review]

**Comment by sekl** [comment]
@mandrigin Thanks for the feedback. I've made changes and rebased, let me know if I missed anything.

**Review by mandrigin — APPROVED** [review]

**Comment by mandrigin** [comment]
@sekl great job 🎉

**Pull request merged**